src/kernel.cr

```diff
@@ -189,23 +189,32 @@ class Process
   def self.after_fork_child_callbacks
     @@after_fork_child_callbacks ||= [
       ->Scheduler.after_fork,
-      ->Event::SignalHandler.after_fork,
-      ->{ Event::SignalChildHandler.instance.after_fork },
+      ->Crystal::SignalHandler.after_fork,
+      ->Crystal::SignalChildHandler.after_fork,
       ->Random::DEFAULT.new_seed,
     ] of -> Nil
   end
 end
 
 {% unless flag?(:win32) %}
-  Signal.setup_default_handlers
-
-  at_exit { Event::SignalHandler.close }
-
   # Background loop to cleanup unused fiber stacks.
   spawn do
     loop do
       sleep 5
       Fiber.stack_pool_collect
     end
   end
+
+  # :nodoc:
+  fun __crystal_sigfault_handler(sig : LibC::Int, addr : Void*)
+    Crystal.restore_blocking_state
```

Comment:
I'm a bit concerned that this call could double fault and the message below will never get printed (there are layers of indirection where another possible fault could be introduced).

Reply:
I didn't touch those lines.

BTW I moved the `LibExt.sefault_sigfault_handler` call from `signal.cr` to `kernel.cr` which felt more in place —this is where we initialize many things— and moved the `fun` along with it but maybe the `fun` should have stayed in `signal.cr`?